### ISSUE TITLE
[Panel] For collapsible variants screen reader can read collapsed panel content.

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3046]
## Current behavior

Screen reader users that interact with collapsed Panel can set screen reader's focus on invisible, collapsed panel content and read it. To do it user have to use "down arrow" in NVDA or "Control + Option + right arrow" in VoiceOver to read inactive elements.

https://github.com/user-attachments/assets/f71dca1d-8726-483d-9f0d-bf97379563d8

https://github.com/user-attachments/assets/f8e18bdd-7d04-4686-9186-0398f748229b

Issue occurs in each collapsible version of Panel which includes: 
1. [Collapsible](https://storybook.sanomalearning.design/?path=/story/layout-panel--collapsible)
2. [Collapsed](https://storybook.sanomalearning.design/?path=/story/layout-panel--collapsed)
3. [No Padding Content](https://storybook.sanomalearning.design/?path=/story/layout-panel--no-padding-content)
4. [Toggle Externally](https://storybook.sanomalearning.design/?path=/story/layout-panel--toggle-externally)
5. [All](https://storybook.sanomalearning.design/?path=/story/layout-panel--all)

## Suggested solution

It's recommended to hide collapsed panel's content from screen reader. Common solution is to use CSS "display: none;" or "visibility: hidden;" as it hides content both visualy and from screen reader.
But other solutions, for example this used in [Accordion Basic](https://storybook.sanomalearning.design/?path=/story/layout-accordion--basic) also should work, as long as screen reader does not announce collapsed content.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
